diff --git a/libcxxabi/src/cxa_guard_impl.h b/libcxxabi/src/cxa_guard_impl.h
index 191a58917..571ee8b8f 100644
--- a/src/cxa_guard_impl.h
+++ b/src/cxa_guard_impl.h
@@ -164,7 +164,7 @@ uint32_t PlatformThreadID() {
 #elif defined(SYS_gettid) && _LIBCPP_HAS_THREAD_API_PTHREAD
 uint32_t PlatformThreadID() {
   static_assert(sizeof(pid_t) == sizeof(uint32_t), "");
-  return static_cast<uint32_t>(syscall(SYS_gettid));
+  return static_cast<uint32_t>(uk_syscall(SYS_gettid));
 }
 #else
 constexpr uint32_t (*PlatformThreadID)() = nullptr;
@@ -428,13 +428,13 @@ void PlatformFutexWake(int* addr) {
 #elif defined(SYS_futex)
 void PlatformFutexWait(int* addr, int expect) {
   constexpr int WAIT = 0;
-  syscall(SYS_futex, addr, WAIT, expect, 0);
+  uk_syscall(SYS_futex, addr, WAIT, expect, 0);
   __tsan_acquire(addr);
 }
 void PlatformFutexWake(int* addr) {
   constexpr int WAKE = 1;
   __tsan_release(addr);
-  syscall(SYS_futex, addr, WAKE, INT_MAX);
+  uk_syscall(SYS_futex, addr, WAKE, INT_MAX);
 }
 #else
 constexpr void (*PlatformFutexWait)(int*, int) = nullptr;
